build/kind/postgres.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: postgres-selfsigned-issuer
+  namespace: open-cluster-management
+spec:
+  selfSigned: {}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgres-external
+  namespace: open-cluster-management
+  labels:
+    app: postgres
+spec:
+  selector:
+    app: postgres
+  ports:
+    - name: postgres-external
+      port: 5432
+      targetPort: 5432
+      nodePort: 30543
+  type: NodePort
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: postgres-cert
+  namespace: open-cluster-management
+spec:
+  dnsNames:
+    - postgres-external.open-cluster-management.svc
+    - postgres-external.open-cluster-management.svc.cluster.local
+    - localhost
+  issuerRef:
+    kind: Issuer
+    name: postgres-selfsigned-issuer
+  secretName: postgres-cert
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: postgres
+  namespace: open-cluster-management
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: postgres
+  template:
+    metadata:
+      labels:
+        app: postgres
+    spec:
+      securityContext:
+        # This is the postgres group.
+        fsGroup: 999
+      containers:
+        - name: postgres
+          command:
+            - docker-entrypoint.sh
+            - -c
+            - ssl=on
+            - -c
+            - ssl_cert_file=/var/lib/postgresql/tls/tls.crt
+            - -c
+            - ssl_key_file=/var/lib/postgresql/tls/tls.key
+            - -c
+            - ssl_ca_file=/var/lib/postgresql/tls/ca.crt
+          # This is a mirror of postgres:13 on Docker Hub to avoid rate limits.
+          image: quay.io/stolostron/grc-ci-postgres:13
+          imagePullPolicy: "IfNotPresent"
+          ports:
+            - containerPort: 5432
+          env:
+            - name: POSTGRES_PASSWORD
+              value: grc
+            - name: POSTGRES_USER
+              value: grc
+            - name: POSTGRES_DB
+              value: ocm-compliance-history
+          volumeMounts:
+            - mountPath: /var/lib/postgresql/data
+              name: postgres-db
+              subPath: data
+            - mountPath: /var/lib/postgresql/tls
+              name: postgres-cert
+              readOnly: true
+      volumes:
+        - name: postgres-cert
+          secret:
+            secretName: postgres-cert
+            # Postgres requires limited permissions on the private key.
+            defaultMode: 0o440
```

Comment:
Is this correct? unusual too me

Reply:
I believe I had a YAML lint error when I didn't use the octal notation. This is the equivalent of `chmod 440` on the files though.